Main.py

```diff
@@ -0,0 +1,70 @@
+import os
+from Ask_gpt import *
+from pptx import Presentation
+
+
+def read_slide(slide):
+    """
+    Retrieves the text content from a slide.
+    :param slide: The slide object from which to extract the text.
+    :return: A string containing the combined text content from the slide's text frames.
+    """
+    text_frames = [shape.text_frame.text for shape in slide.shapes if shape.has_text_frame]
+    return '\n'.join(text_frames)
+
+
+def process_slide(i, slide):
+    """
+    Processes a slide by writing its content to a file after a delay and obtaining a response using external processing.
+    :param i: The index of the slide.
+    :param slide: The text content of the slide
+    :return:   The processed slide content as a string.
+    """
+    processed_content = "Page " + str(i) + "\n\n"
+    time.sleep(20)
```

Comment:
I didn't understand why you sleep for 20 seconds, it is a lot of time! 
you should try to make this code faster instead of slower